apps/api/src/services/vote-manager.service.ts

```diff
@@ -287,6 +316,29 @@ export class VoteManager {
       throw error;
     }
 
+    // Check voting dates if they are set
+    const now = new Date();
+
+    // If voting start date is set and we haven't reached it, voting is not allowed
+    if (competition.votingStartDate && now < competition.votingStartDate) {
```

Comment:
optional: since both methods make a similar date check, it might be helpful for a helper like `isVotingAllowedByDates`...not essential tho.